zebedee-cms/src/main/java/com/github/onsdigital/zebedee/api/Collections.java

```diff
@@ -96,4 +96,103 @@ public int compare(CollectionDescription o1, CollectionDescription o2) {
         }
         return null;
     }
+
+    /**
+     * Put endpoints for /collections.
+     * This supports only /collections/{collection_id}/instances/{instance_id}
+     */
+    @PUT
+    public void put(HttpServletRequest request, HttpServletResponse response) throws ZebedeeException, IOException {
+
+        Session session = zebedeeCmsService.getSession(request);
+        if (!zebedeeCmsService.getPermissions().canEdit(session)) {
+            logInfo("Forbidden request made to the collection endpoint").log();
+            response.setStatus(HttpStatus.SC_FORBIDDEN);
+            return;
+        }
+
+        Path path = Path.newInstance(request);
+        List<String> segments = path.segments();
+
+        // /collections/{collection_id}/instances/{instance_id}
+        if (segments.size() < 4 ||
+                !segments.get(2).equalsIgnoreCase("instances")) {
+
+            logInfo("Endpoint for colletions not found").addParameter("path", path).log();
+            response.setStatus(HttpStatus.SC_NOT_FOUND);
+            return;
+        }
+
+        String collectionID = segments.get(1);
+        String instanceID = segments.get(3);
+
+        logInfo("PUT called on /collections/{collection_id}/instances/{instance_id} endpoint")
+                .addParameter("collectionID", collectionID)
+                .addParameter("instanceID", instanceID)
+                .log();
+
+        datasetService.addInstanceToCollection(collectionID, instanceID);
+    }
+
+    /**
+     * Delete endpoints for /collections.
+     * This supports only /collections/{collection_id}/instances/{instance_id}
+     */
+    @DELETE
+    public void delete(HttpServletRequest request, HttpServletResponse response) throws ZebedeeException, IOException {
+
+        Session session = zebedeeCmsService.getSession(request);
+        if (!zebedeeCmsService.getPermissions().canEdit(session)) {
+            logInfo("Forbidden request made to the collection endpoint").log();
+            response.setStatus(HttpStatus.SC_FORBIDDEN);
+            return;
+        }
+
+        Path path = Path.newInstance(request);
+        List<String> segments = path.segments();
+
+        // /collections/{collection_id}/instances/{instance_id}
+        if (segments.size() < 4 ||
```

Comment:
Pull this uri logic out into a separate method and then call it from these methods

Reply:
👍